### DISCUSSION TITLE
Ability to make country display private

### DISCUSSION THREAD

**Discussion by sharphello-world** [ppy/osu#24560] [Ideas]
There should be a setting on https://osu.ppy.sh/home/account/edit below "country" that lets you hide the country that you made your account in so that people can't tell where you're from or where you made your account by looking at your profile, both in the web view and in-game (including multiplayer games). A profile will still have a country associated with it but it's just not publicly viewable.

Here are the problems someone might face:

* I made my account when I was in a foreign country. I'm no longer there. Now in multiplayer games I keep getting talked to by people in a language I don't speak.
* I made my account when I was in a foreign country. I'm no longer there. I don't want to be associated with that country.
* I am from the country that my osu! profile uses but I don't want to be talked to by people from my own country.
* I don't want people to assume things about me based on this country flag.
* I simply don't want people to know where I'm from or where I made my account!

osu! doesn't even let you change your country so at least let me hide it!
It used to be possible to change your country but this service was shut down and I can see why that might have been shut down but providing this option to just hide the country from your profile should satisfy the crowd that wants their country changed at least a little bit.

Many other video games either don't show your country anyway or at least have an option to hide it.

The game does not need the country for anything except:

* Tournaments: in this case just require participants to make their country public. Most people don't participate in tournaments.
* Country leaderboards: the solution is to not include people that have their countries hidden in country-based leaderboards. I might not want to participate in them anyway. This also hides "Country Ranking" from my profile.
* Am I missing something?

This affects both https://github.com/ppy/osu and https://github.com/ppy/osu-web.

**Comment by peppy** [comment]
https://osu.ppy.sh/home/news/2023-07-28-enabling-country-flag-changes

Also are you able to provide your account link so I can gain some insight? This is the second time I've seen a request for this in 16 years.

**Comment by Mosswool** [comment]
I would also very much like this to be a feature.

In another significant online game service, chess.com, easy flag changes and an anonymous "International" flag are allowed.

I think many users would benefit from Osu! following this example.

**Reply by Tenexxt** [reply]
I think the current system for countries is fine in osu, its like this because you AREN'T supposed to change it manually. The reason you can't just change your country like in chess (with 2 clicks) is mostly the same as for why you can't change your username often

If you are in a country for a longer period, there is always https://osu.ppy.sh/home/news/2023-07-28-enabling-country-flag-changes

As for hiding the country, its not a bad idea i guess - the points in this post aren't bad, tho i doubt this feature would be usefull for many its still not a bad idea if more people will want it.

**Reply by nagi-desuuu** [reply]
For the "international" flag, this exists now, which should resolve that part of the problem.
![image](https://github.com/user-attachments/assets/8867b4ce-b793-4abb-a51f-5c9686d0ca85)

**Reply by Tenexxt** [reply]
> For the "international" flag, this exists now, which should resolve that part of the problem. ![image](https://private-user-images.githubusercontent.com/75596775/369986299-8867b4ce-b793-4abb-a51f-5c9686d0ca85.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcxMTQxOTMsIm5iZiI6MTcyNzExMzg5MywicGF0aCI6Ii83NTU5Njc3NS8zNjk5ODYyOTktODg2N2I0Y2UtYjc5My00YWJiLWE1MWYtNWM5Njg2ZDBjYTg1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIzVDE3NTEzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUzZmFmZjA4YjY3ODZlNWMwNTg5ZmM3ZGVjOWZlY2JmZmY3NjZkZTk2ZjYyMTdjZTc5ZjcwMDE0MmM2ZDZiNzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.tTvnfQBgeUhSgzztE2MXLWICT1IdpFXGZ_iZ7qosYm4)

This option is useless and doesn't solve the "problem" IMO. The names of countries are still displayed when you hover, and on the side of the international flag, so this doesn't hide the country at all. Also this wouldn't hide it everywhere, like its requested here, because it only hides it in-game, and once again - it doesn't actualy hide the anyones country.

**Comment by vmx2f** [comment]
+1 for this please, there's no way we can't hide our location. 

I found a thread that has +465 upvotes since 5 years ago and this feature hasn't been considered. https://osu.ppy.sh/community/forums/topics/1236711

<img width="750" height="326" alt="please" src="https://github.com/user-attachments/assets/c18fadab-da4d-435b-bb1e-6b9e3aac2663" />
